Contents/WeaponModification/WeaponModificationUI.cs

```diff
@@ -348,6 +296,94 @@ public ImprovisedUITextBox(string text, float textScale = 1f, bool large = false
 			}
 		}
 	}
+	/// <summary>
+	/// This is the UI where we will handle all the stupid logic<br/>
+	/// it should only be working on client side only and always load so make sure to use TryGet"X" here<br/>
+	/// I will work with UI state as that allow us to change state of UI whenever active and deactive
+	/// </summary>
+	public class WeaponModificationUI : UIState {
+		public int whoAmI = -1;
+		Player player;
+		public override void OnActivate() {
+			Elements.Clear();
+			if (whoAmI == -1)
+				return;
+			Player player = Main.player[whoAmI];
```

Comment:
I'm 100% sure UI ever ever runs just for client player, so just on `Main.LocalPlayer`. So most of this part of code can be simplified. (unless you have a whatever good reason)